Add Pylint document (#432)
sider/sider-docs: docs/tools/python/pylint.md

---
id: pylint
title: Pylint
sidebar_label: Pylint
hide_title: true
---

# Pylint

| Supported Version | Language     | Website                         |
| ----------------- | ------------ | ------------------------------- |

> | 2.5.2             | Python 3.8.2 | https://www.pylint.org/ |

Reviewer: [suggestion] <https://www.pylint.org/> seems unmaintainable. How about changing it as follows?

```suggestion
| 2.5.2             | Python 3.8.2 | https://pylint.pycqa.org/en/stable/index.html |
```

Author: I think https://www.pylint.org/ is maintained, because there is `Read the doc` button and the page is the latest version of the doc (http://pylint.pycqa.org/en/latest/).

Reviewer: When I click `Read the doc` button, the page is redirected to docs.pylint.org and pylint.pycqa.org and then it is not shown. 😢 

<img width="1059" alt="image" src="https://user-images.githubusercontent.com/473530/82998010-1e9b4780-a042-11ea-986a-7ecde8622cc7.png">

I can view <https://pylint.pycqa.org/en/latest/> (HTTPS).

Reviewer: I think we should provide <https://pylint.pycqa.org/en/stable/> link instead of `latest`. Because we provide a stable version of Pylint.

Reviewer: Here is a shortened version:

```suggestion
| 2.5.2             | Python 3.8.2 | https://pylint.pycqa.org/en/stable/ |
```

**Pylint** is a Python static code analysis tool which looks for programming errors, helps to enforce a coding standard, sniffs for code smells and offers simple refactoring suggestions.

## Getting Started

To start using Pylint, enable it in your [repository settings](../../getting-started/repository-settings.md).

> To customize Pylint, put a `.pylintrc` file in your repository.

Reviewer: [suggestion] How about adding a code sample to generate a file?

---

To customize Pylint, put a `.pylintrc` file in your repository. For example:

```console
$ pylint --generate-rcfile > .pylintrc
```

## Configuration

You can customize the analysis via `sider.yml`:

```yaml
linter:

>   pylint:

Reviewer: [suggestion] Please add `target`, for example:

```suggestion
  pylint:
    target:
      - "src/**/*.py"
```

    rcfile: ./folder/.pylintrc
    errors-only: true
    ignore:
      - bad1.py
      - bad2.py
```

| Name                                                                                  | Type                 | Default    |
| ------------------------------------------------------------------------------------- | -------------------- | ---------- |
| [`rcfile`](#rcfile)                                                                   | `string`             | -          |

> | [`target`](#target)                                                                   | `string`, `string[]` | `**/*.{py}`|

Reviewer: Please move the `target` option to the top. This is a Sider-specific option.

| [`errors-only`](#errors-only)                                                         | `boolean`            | -          |
| [`ignore`](#ignore)                                                                   | `string`, `string[]` | -          |

### `rcfile`

This option allows you to specify a configuration file you want.

See also the [`--rcfile`](http://pylint.pycqa.org/en/latest/user_guide/run.html?highlight=rcfile#command-line-options) option.

```yaml
linter:
  pylint:
    rcfile: ./folder/.pylintrc
```

> ### `target`

Reviewer: Please move this section to the top of the other sections, too.

This option allows you to specify files or directories to analyze via Pylint. If you specify multiple targets, configure as follow:

```yaml
linter:
  pylint:
    target:
      - ./folder/bad1.py
      - ./folder/bad2.py
```

### `errors-only`

> This option allows you to show only pylint error messags.

Reviewer: Please fix the typos:

```suggestion
This option allows you to show only Pylint's error messages.
```

---
id: pylint
title: Pylint
sidebar_label: Pylint (beta)
hide_title: true
---

# Pylint

> This is **BETA**. The behavior of this tool might change.

| Supported Version | Language     | Website                              |
| ----------------- | ------------ | ------------------------------------ |
| 2.5.2             | Python 3.8.2 | https://pylint.pycqa.org/en/stable/  |

**Pylint** is a Python static code analysis tool which looks for programming errors, helps to enforce a coding standard, sniffs for code smells and offers simple refactoring suggestions.

## Getting Started

To start using Pylint, enable it in your [repository settings](../../getting-started/repository-settings.md).

To customize Pylint, put a `.pylintrc` file in your repository.

If you don't have `.pylintrc` file, you can make it by executing the command below.

```bash
pylint --generate-rcfile > .pylintrc
```

## Configuration

You can customize the analysis via `sider.yml`:

```yaml
linter:
  pylint:
    target:
      - ./folder/bad1.py

>       - ./folder/bad2.py

Reviewer: Please add a glob example as follows:

```suggestion
      - "./**/*test.py"
```

    rcfile: ./folder/.pylintrc
    errors-only: true
    ignore:
      - bad1.py
      - bad2.py
```

| Name                                                                                  | Type                 | Default    |
| ------------------------------------------------------------------------------------- | -------------------- | ---------- |
| [`target`](#target)                                                                   | `string`, `string[]` | `**/*.{py}`|
| [`rcfile`](#rcfile)                                                                   | `string`             | -          |
| [`errors-only`](#errors-only)                                                         | `boolean`            | -          |
| [`ignore`](#ignore)                                                                   | `string`, `string[]` | -          |

### `target`

> This option allows you to specify files or directories to analyze via Pylint. If you specify multiple targets, configure as follows:

Reviewer: How about adding a description as follows?

```suggestion
This option allows you to specify files or directories to analyze via Pylint. Glob format is available.
If you specify multiple targets, configure as follows:
```

```yaml
linter:
  pylint:
    target:
      - ./folder/bad1.py

>       - ./folder/bad2.py

Reviewer: Same here:

```suggestion
      - "./**/*test.py"
```

```

### `rcfile`

> This option allows you to specify a configuration file you want.

Reviewer: [question] If a document about Pylint configuration file (e.g. `.pylintrc`) exist, the following change may be useful.

```suggestion
This option allows you to specify a [Pylint configuration file](https://pylint.pycqa.org/en/stable/...) you want.
```

But, does such a document actually exist?

Author: I tried to find it on the document. I couldn't find it. 😭

> See also the [`--rcfile`](http://pylint.pycqa.org/en/latest/user_guide/run.html?highlight=rcfile#command-line-options) option.

Reviewer: Please remove the description. It looks needless and some users cannot see the page  <http://pylint.pycqa.org/en/latest> (like me).
